core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -153,30 +194,38 @@ trait SubmissionComponent {
 
     def listWithSubmitter(workspaceContext: Workspace): ReadWriteAction[Seq[SubmissionListResponse]] = {
       val query = for {
-        (submissionRec, entityRec) <- findByWorkspaceId(
+        submissionRec <- findByWorkspaceId(
           workspaceContext.workspaceIdAsUUID
-        ) joinLeft entityQuery on (_.submissionEntityId === _.id)
+        )
         methodConfigRec <- methodConfigurationQuery if submissionRec.methodConfigurationId === methodConfigRec.id
-      } yield (submissionRec, methodConfigRec, entityRec)
+      } yield (submissionRec, methodConfigRec)
 
       for {
         workflowStatusResponses <- GatherStatusesForWorkspaceSubmissionsQuery.gatherWorkflowIdsAndStatuses(
           workspaceContext.workspaceIdAsUUID
         )
         states = workflowStatusResponses.groupBy(_.submissionId)
         recs <- query.result
-      } yield recs.map { case (submissionRec, methodConfigRec, entityRec) =>
+        submissionIds = recs.map(_._1.id).toSeq
+        subTuples <- submissionEntityQuery.getEntitiesForSubmissions(submissionIds)
+        submissionGroup = recs.map { case (submissionRec, methodConfigRec) =>
+          val entityRecs = subTuples.find(_._1 == submissionRec.id).map(_._2).getOrElse(Seq.empty)
+          (submissionRec, methodConfigRec, entityRecs)
+        }
+      } yield submissionGroup.map { case (submissionRec, methodConfigRec, entityRecs) =>
         val config = methodConfigurationQuery.unmarshalMethodConfig(methodConfigRec, Map.empty, Map.empty)
         val statusCounts = states
           .getOrElse(submissionRec.id, Seq.empty)
           .map(x => Map(x.workflowStatus -> x.count))
           .foldLeft(Map.empty[String, Int])(_ |+| _)
         val maybeWorkflowIds = states.getOrElse(submissionRec.id, Seq.empty).flatMap(_.workflowId).sorted
         val workflowIds = if (maybeWorkflowIds.nonEmpty) Some(maybeWorkflowIds) else None
-        SubmissionListResponse(unmarshalSubmission(submissionRec, config, entityRec.map(_.toReference), Seq.empty),
-                               workflowIds,
-                               statusCounts,
-                               config.deleted
+        val singleEntity = if (entityRecs.size == 1) Some(entityRecs.head) else None // TODO smarten this up
```

Comment:
If we don't return the singleEntity when unmarshalling, existing code relying on it might break, so I've included it even though it's a part of the entityRecs.